public/app/features/explore/state/actions.ts

```diff
@@ -698,22 +716,29 @@ const togglePanelActionCreator = (
     | ActionCreator<ToggleTablePayload>
 ) => (exploreId: ExploreId) => {
   return (dispatch, getState) => {
-    let shouldRunQueries;
-    dispatch(actionCreator({ exploreId }));
-    dispatch(stateSave());
+    let shouldRunQueries, uiFragmentStateUpdate: Partial<ExploreUIState>;
 
     switch (actionCreator.type) {
       case toggleGraphAction.type:
-        shouldRunQueries = getState().explore[exploreId].showingGraph;
+        const isShowingGraph = getState().explore[exploreId].showingGraph;
+        shouldRunQueries = !isShowingGraph;
+        uiFragmentStateUpdate = { showingGraph: !isShowingGraph };
```

Comment:
Always a bit annoying if the action creators need to know the state shape.

Reply:
Agree, this should be a part of the part of the payload.

Reply:
@davkal Changes in #cee2e47